sam-app/tests/unit/test_handler.py

```diff
@@ -81,16 +81,283 @@ def apigw_event():
+        "StatisticValues": {
+            "SampleCount": 12.17,
+            "Sum": 12.17,
+            "Minimum": 12.17,
+            "Maximum": 12.17
+        },
+        "Unit": "Seconds",
+        "StorageResolution": 12
+    }
+
+@pytest.fixture()
+def event7():
+    return {
+        "Namespace": "theNamespace",
+        "MetricName": "theMetricName",
+        "Dimensions": [
+            {
+                "Name": "theName",
+                "Value": "theValue"
+            }
+        ],
+        "Timestamp": "Fake time stamp",
+        "Value": 12.17,
+        "StatisticValues": {
+            "SampleCount": 12.17,
+            "Sum": 12.17,
+            "Minimum": 12.17,
+            "Maximum": 12.17
+        },
+        "Unit": "Seconds",
+        "StorageResolution": 12
+    }
+
+@pytest.fixture()
+def event8():
+    return {
+        "Namespace": "theNamespace",
+        "MetricName": "theMetricName",
+        "Dimensions": [
+            {
+                "Name": "theName",
+                "Value": "theValue"
+            }
+        ],
+        "Timestamp": "1528319852774",
+        "Value": "Fake Value",
+        "StatisticValues": {
+            "SampleCount": 12.17,
+            "Sum": 12.17,
+            "Minimum": 12.17,
+            "Maximum": 12.17
+        },
+        "Unit": "Seconds",
+        "StorageResolution": 12
+    }
+
+@pytest.fixture()
+def event9():
+    return {
+        "Namespace": "theNamespace",
+        "MetricName": "theMetricName",
+        "Dimensions": [
+            {
+                "Name": "theName",
+                "Value": "theValue"
+            }
+        ],
+        "Timestamp": "1528319852774",
+        "Value": 12.17,
+        "StatisticValues": {
+            "SampleCount": 12.17,
+            "Sum": 12.17,
+            "Minimum": 12.18,
+            "Maximum": 12.17
+        },
+        "Unit": "Seconds",
+        "StorageResolution": 12
+    }
+
+@pytest.fixture()
+def event10():
+    return {
+        "Namespace": "theNamespace",
+        "MetricName": "theMetricName",
+        "Dimensions": [
+            {
+                "Name": "theName",
+                "Value": "theValue"
+            }
+        ],
+        "Timestamp": "1528319852774",
+        "Value": 12.17,
+        "StatisticValues": {
+            "SampleCount": 12.17,
+            "Sum": 12.17,
+            "Minimum": 12.17,
+            "Maximum": 12.17
+        },
+        "Unit": "seconds",
+        "StorageResolution": 12
+    }
+
+"""
+def test_lambda_handler_provided(apigw_event):
+
+    ret = app.lambda_handler_provided(apigw_event, "")
     assert ret['statusCode'] == 200
 
     for key in ('message', 'location'):
         assert key in ret['body']
 
     data = json.loads(ret['body'])
     assert data['message'] == 'hello world'
+"""
+
+def test_lambda_handler1(event1):
+    ret = app.lambda_handler(event1, "")
+    assert ret == "Function executed through to completion."
+
+def test_lambda_handler2(event2):
```

Comment:
These tests are super hard to read and the numbers after each name tell me nothing of what scenario is being tested. I'll send you example unit tests so you can see what they should look like.